IdentityCore/src/cache/serializers/MSIDKeyedArchiverSerializer.m

```diff
@@ -23,17 +23,29 @@
 
 #import "MSIDKeyedArchiverSerializer.h"
 #import "MSIDToken.h"
+#import "MSIDUserInformation.h"
+
+@interface MSIDKeyedArchiverSerializer ()
+
+@end
 
 @implementation MSIDKeyedArchiverSerializer
 
+#pragma mark - MSIDTokenSerializer
+
 - (NSData *)serialize:(MSIDToken *)token
 {
-    return nil;
+    return [NSKeyedArchiver archivedDataWithRootObject:token];
 }
 
 - (MSIDToken *)deserialize:(NSData *)data
 {
-    return nil;
+    NSKeyedUnarchiver *unarchiver = [[NSKeyedUnarchiver alloc] initForReadingWithData:data];
+    [unarchiver setClass:MSIDUserInformation.class forClassName:@"ADUserInformation"];
+    MSIDToken *token = [unarchiver decodeObjectOfClass:MSIDToken.class forKey:NSKeyedArchiveRootObjectKey];
```

Comment:
why is this necessary? Isn't NSKeyedUnarchiver setClass:forClassName: enough?

Reply:
If we set class name globally in "NSKeyedUnarchiver setClass:forClassName:" it will conflict with archiving ADUserInformation in ADAL. Here we do manipulation with name locally so it will not affect other archivers.

Reply:
I see, but do we still want to archive ADUserInformation in ADAL?

Reply:
We will not need to archive ADUserInformation anymore, but I still think it is safer to do name manipulation locally. What advantages will we get from doing it globally?